src/main/java/com/internship/socialnetwork/controller/PostController.java

```diff
@@ -0,0 +1,54 @@
+package com.internship.socialnetwork.controller;
+
+import com.internship.socialnetwork.dto.PostDto;
+import com.internship.socialnetwork.service.PostService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/posts")
+@RestController
+public class PostController {
+
+    private final PostService postService;
+
+    @GetMapping
+    public ResponseEntity<List<PostDto>> getAll() {
+        return ResponseEntity.ok(postService.getAll());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<PostDto> getOne(@RequestParam Long id) {
+        return ResponseEntity.ok(postService.get(id));
+    }
+
+    //ToDo: Create getAllPostsById in the user controller
+
+    @PostMapping
+    public ResponseEntity<PostDto> create(@Valid @RequestBody PostDto postDto) {
+        return ResponseEntity.status(201).body(postService.create(postDto));
```

Comment:
You don't have to hardcode 201, use this instead: ResponseEntity.status(HttpStatus.CREATED)